### DISCUSSION TITLE
How do I determine if no rows found is caused by RLS or incorrect id?

### DISCUSSION THREAD

**Discussion by milesingrams** [supabase/supabase#8114] [Questions] [answered]
On the frontend I want to be able to request a DB entry and display an error if the user does not have RLS permissions to select that entry. But the same error is returned if I request an entry that doesn't exist:

```
{"message":"JSON object requested, multiple (or no) rows returned","code":"PGRST116","details":"Results contain 0 rows, application/vnd.pgrst.object+json requires 1 row","hint":null}
```

Is there any way to differentiate these two cases?

**Comment by GaryAustin1** [comment, marked as answer]
No.  RLS is like an additional where clause on your query.  Basically if you don't meet RLS, then you don't see the rows at all is the intent, not that you get an error on access. 
This has a bit of a comment on it.
https://dba.stackexchange.com/questions/300776/postgres-rls-how-to-error-instead-of-ignoring-rows-not-matched-by-the-policy

**Reply by milesingrams** [reply]
Good to know and thanks for the info! I didn't realize it provides a vector for attack to let the end user know a document exists but is inaccessible at that location but I guess it makes sense.
